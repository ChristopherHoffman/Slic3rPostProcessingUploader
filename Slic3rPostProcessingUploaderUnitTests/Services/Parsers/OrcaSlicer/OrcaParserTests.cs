using Slic3rPostProcessingUploader.Services.Parsers.AnycubicSlicerNext;
using Slic3rPostProcessingUploader.Services.Parsers.OrcaSlicer;
using Slic3rPostProcessingUploaderUnitTests.Services.Parsers.AnycubicSlicerNext;
using Snapshooter.MSTest;


namespace Slic3rPostProcessingUploaderUnitTests.Services.Parsers.OrcaSlicer
{
    [TestClass]
    public class OrcaParserTests
    {
        [TestMethod]
        public void ShouldReturnAnEmptySettingWhenGcodeIsEmpty()
        {
            var parser = new OrcaParser("");
            var result = parser.ParseGcode("");
            Assert.IsNotNull(result);

            Snapshot.Match(result, matchOptions => matchOptions.HashField("settings.Snapshot"));
        }

        [TestMethod]
        public void ShouldReturnExpectedValuesWhenGivenFullGcode()
        {
            var parser = new OrcaParser("");
            var result = parser.ParseGcode(OrcaParserTestGcode.CalibrationCube);

            Snapshot.Match(result, matchOptions => matchOptions.HashField("settings.Snapshot"));
        }

        [TestMethod]
        public void ShouldRenderTheExpectedNoteWhenGivenATemplateWithNoReplacements()
        {
            string template = "Settings:";

            var parser = new OrcaParser(template);
            var result = parser.ParseGcode(OrcaParserTestGcode.CalibrationCube);

            Assert.AreEqual("Settings:", result.settings.note);
        }

        [TestMethod]
        public void ShouldRenderTheExpectedNoteWhenGivenATemplateWithASingleReplacement()
        {
            string template = """
                Settings:
                    Layer Height: {{layer_height}}
                """;

            var parser = new OrcaParser(template);
            var result = parser.ParseGcode(OrcaParserTestGcode.CalibrationCube);

            Assert.AreEqual("""
                Settings:
                    Layer Height: 0.2
                """, result.settings.note);
        }

        [TestMethod]
        public void ShouldRenderTheExpectedNoteWhenGivenATemplateWithMultipleReplacements()
        {
            string template = """
                Settings:
                    Layer Height: {{layer_height}}
                    First Layer Height: {{first_layer_height}}
                    Wall Loops: {{wall_loops}}
                    Top Shell Layers: {{top_shell_layers}}
                    Bottom Shell Layers: {{bottom_shell_layers}}
                    Sparse Infill Density: {{sparse_infill_density}}
                """;

            var parser = new OrcaParser(template);
            var result = parser.ParseGcode(OrcaParserTestGcode.CalibrationCube);

            Assert.AreEqual("""
                Settings:
                    Layer Height: 0.2
                    First Layer Height: 0.200
                    Wall Loops: 3
                    Top Shell Layers: 4
                    Bottom Shell Layers: 3
                    Sparse Infill Density: 15%
                """, result.settings.note);
        }

        [TestMethod]
        public void ShouldRenderFullTemplateWhenGivenAGcodeWithTwoFilaments()
        {
            var parser = new OrcaParser("");
            var result = parser.ParseGcode(OrcaParserTestGcode.CalibrationCubeTwoFilament);

            Snapshot.Match(result, matchOptions => matchOptions.HashField("settings.Snapshot"));
        }

    }
}
