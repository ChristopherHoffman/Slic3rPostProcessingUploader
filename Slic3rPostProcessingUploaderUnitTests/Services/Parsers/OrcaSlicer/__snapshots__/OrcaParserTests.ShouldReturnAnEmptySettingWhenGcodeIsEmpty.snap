{
  "Slicer": "OrcaSlicer",
  "CuraVersion": "Unknown",
  "PluginVersion": null,
  "settings": {
    "note": "Settings:\n\nQuality:\n  Layer Height:\n    Layer Height: \n    First Layer Height: \n  Line Width:\n    Default Line Width: \n  Seam:\n    Seam Position: \n  Precision:\n    Precise Wall: \n  Ironing:\n    Ironing: \n  Wall Generator:\n    Generator: \n  Walls And Surfaces:\n    Only One Wall on Top Surfaces: \n\nStrength:\n  Walls:\n    Wall Loops: \n  Top/Bottom Shells:\n    Top Shell Layers: \n    Top Shell Thickness: \n    Bottom Shell Layers: \n    Bottom Shell Thickness: \n  Infill:\n    Sparse Infill Density: \n    Sparse Infill Pattern: \n\nSpeed:\n  First Layer Speed:\n    First Layer: \n    Number of Slow Layers: \n  Other Layers Speed:\n    Outer Wall: \n    Inner Wall: \n    Sparse Infill: \n    Internal Solid Infill: \n    Top Surface: \n  Travel Speed:\n    Travel: \n\nSupport:\n  Support:\n    Enable Support: \n    Type: \n    Style: \n    Threshold Angle: \n    On Build Plate Only: \n  Raft:\n    Raft Layers: \n\nMultimaterial:\n  Prime Tower:\n    Enable: \n    Width: \n    Brim Width: \n  Ooze Prevention:\n    Enable: \n  Flush Options:\n    Flush into Objects' Infill: \n    Flush into Objects' Support: \n  Advanced:\n    Use Beam Interlocking: \n\nOther:\n  Skirt:\n    Skirt Type: \n    Skirt Loops: \n    Skirt Min Extrusion Length: \n  Brim:\n    Brim Type: \n    Brim Width: \n  Special Mode:\n    Slicing Mode: \n    Print Sequence: \n    Spiral Vase: \n    Fuzzy Skin: ",
    "print_name": null,
    "time_stamp": null,
    "estimated_print_time_seconds": 0,
    "material_used_mg": 0,
    "Snapshot": null,
    "file_name": null,
    "filamentUsage": []
  }
}
